### ISSUE TITLE
Mirror Shadowsocks clients error

### ISSUE THREAD

**Issue by juneyao** [jlund/streisand#222]
The installation stopped at:

TASK: [streisand-mirror | Mirror Shadowsocks clients] ************************* 
changed: [xx.xx.xx.xx] => (item={'url': u'https://github.com/shadowsocks/shadowsocks-android/releases/download/v2.6.5/shadowsocks-nightly-2.6.5.apk', 'checksum': u'16907bcb5c0e51a59fdf6e41c49f0a6265e8dece8b9bb811df98f8d3aaa98233'})
failed: [xx.xx.xx.xx] => (item={'url': u'http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/Shadowsocks-win-dotnet4.0-2.3.1.zip', 'checksum': u'd63360b413819f557acab9080718a84c72d5d9768ecccb3e6747408ce05de5a1'}) => {"dest": "/var/www/streisand/mirror/shadowsocks", "failed": true, "gid": 33, "group": "www-data", "item": {"checksum": "d63360b413819f557acab9080718a84c72d5d9768ecccb3e6747408ce05de5a1", "url": "http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/Shadowsocks-win-dotnet4.0-2.3.1.zip"}, "mode": "0755", "owner": "www-data", "response": "Request failed: <urlopen error [Errno 101] Network is unreachable>", "size": 4096, "state": "directory", "status_code": -1, "uid": 33, "url": "http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/Shadowsocks-win-dotnet4.0-2.3.1.zip"}
msg: Request failed
failed: [xx.xx.xx.xx] => (item={'url': u'http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/ShadowsocksX-2.6.3.dmg', 'checksum': u'f0e263dd1d74b0b6977389f2b8b28c524bceccbc5ad24ca8a8164b92ede1c45a'}) => {"dest": "/var/www/streisand/mirror/shadowsocks", "failed": true, "gid": 33, "group": "www-data", "item": {"checksum": "f0e263dd1d74b0b6977389f2b8b28c524bceccbc5ad24ca8a8164b92ede1c45a", "url": "http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/ShadowsocksX-2.6.3.dmg"}, "mode": "0755", "owner": "www-data", "response": "Request failed: <urlopen error [Errno 101] Network is unreachable>", "size": 4096, "state": "directory", "status_code": -1, "uid": 33, "url": "http://iweb.dl.sourceforge.net/project/shadowsocksgui/dist/ShadowsocksX-2.6.3.dmg"}
msg: Request failed
changed: [xx.xx.xx.xx] => (item={'url': u'https://pypi.python.org/packages/source/s/shadowsocks/shadowsocks-2.4.3.tar.gz', 'checksum': u'a38954c7ac3497c6796c1abb96b04bc705bf96039f79ab91fa8dc41607ed74f8'})

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/streisand/amazon.retry

127.0.0.1                  : ok=17   changed=5    unreachable=0    failed=0   
xx.xx.xx.xx             : ok=120  changed=105  unreachable=0    failed=1   

sourceforge and github links seem outdated.

are they necessary? how much of the installation was left?

**Comment by jlund** [comment]
Thanks for the bug report. It looks like the Shadowsocks team have switched from SourceForge to GitHub to host their client downloads, which is a good move. I updated each of the mirrored clients to its latest version and download location, and this issue is fixed.

The mirror step is the second-to-last set of tasks that get executed, so the server setup was almost complete. Mirroring these files is critically important for users in countries that block access to the standard client download locations (e.g. China).

You can just delete the partially configured server, do a `git pull` to grab the latest commit, run the setup script again, and you will be all set. Sorry about the trouble!

**Comment by juneyao** [comment]
I am sorry it took me long to get this working.
I had to revert the git pull and manually update the links for the installation to complete.. After the git pull i was stuck at "failed to create ec2 security group"..  I dont know why
Installation completed on u14.04 - u14.04. previously, I had to give up jessie-jessie. On Jessie-jesie I got stuck at creating the ec2 instance (I am sorry I do not have logs)
I was always using VirtualBox

Now finally reading the generated docs, I am impressed and very thankful!
